e2e/cypress/integration/team_settings/teams_spec.js

```diff
@@ -211,4 +220,175 @@ describe('Teams Suite', () => {
+
+        // # Click on team settings menu item
+        cy.findByText('Team Settings').should('be.visible').click();
+
+        // # Click on the team name menu item
+        cy.findByText('Team Name').should('be.visible').click();
+
+        // # Change team name in the input
+        cy.get('#teamName').should('be.visible').clear().type('Test team');
+
+        // Save new team name
+        cy.get('#saveSetting').click();
+
+        // # Close the team settings
+        cy.get('body').type('{esc}', {force: true});
+
+        // Team display name shows as "Testing Team" at top of main menu
+        cy.get('#headerTeamName').invoke('text').should('eq', 'Test team');
+
+        // Team initials show in the team icon in the sidebar
+        cy.get(`#${testTeam.name}TeamButton`).scrollIntoView().should('have.attr', 'aria-label', 'test team team');
+
+        // # Verify url has original team name
+        cy.url().should('include', `/${testTeam.name}/channels/town-square`);
+    });
+
+    it('MM-T2317 Team setting / Update team description', () => {
+        // # visit /
+        cy.visit(`/${testTeam.name}/channels/town-square`);
+
+        // # Open the hamburger menu
+        cy.findByLabelText('main menu').should('be.visible').click();
+
+        // # Click on team settings menu item
+        cy.findByText('Team Settings').should('be.visible').click();
+
+        // # Click on the team description menu item
+        cy.findByText('Team Description').should('be.visible').click();
+
+        // # Change team description in the input
+        cy.get('#teamDescription').should('be.visible').clear().type('This is the best team');
+        cy.get('#teamDescription').should('have.value', 'This is the best team');
+
+        // Save new team description
+        cy.get('#saveSetting').click();
+
+        // # Close the team settings
+        cy.get('body').type('{esc}', {force: true});
+
+        // Team tooltip should show description
+        verifyToolTip('#headerTeamName', 'This is the best team');
+    });
+
+    it('MM-T2318 Allow anyone to join this team', () => {
+        // # UI
+
+        cy.visit(`/${testTeam.name}/channels/town-square`);
+
+        // # Open the hamburger menu
+        cy.findByLabelText('main menu').should('be.visible').click();
+
+        // # Click on team settings menu item
+        cy.findByText('Team Settings').should('be.visible').click();
+
+        // # Click on the team description menu item
+        cy.findByText('Allow any user with an account on this server to join this team').should('be.visible').click();
+
+        // # Change team description in the input
+        cy.get('#teamOpenInvite').click();
+
+        // Save new team description
+        cy.get('#saveSetting').click();
+
+        // # Close the team settings
+        cy.get('body').type('{esc}', {force: true});
+
+        // # Functionality
+
+        cy.visit('/login');
+        cy.apiLogin(newUser);
```

Comment:
Please do login via API first before visiting a URL, like
```suggestion
        // # Login as new user
        cy.apiLogin(newUser);
        cy.visit('/');
```